### PR TITLE
Compatibility with jfx19

### PR DESCRIPTION
**Pull request by Siedlerchr**
I simply renamed them with the suffix Opt

### CODE REVIEW

**Comment by calixtus** [comment]
Any chance this is getting merged soon. @tobiasdiez ? Java 19 will be released in 2 weeks...

**Comment by Siedlerchr** [comment]
@calixtus  jfx != jdk 19 They are independent. Nonetheless, jfx19 will be released in September

**Comment by tobiasdiez** [comment]
The rename looks good to me, but it needs a bit more cleanup before we can merge it. In particular,
- Change the documentation (in particular, readme)
- Discuss the difference of the easybind methods to the new jfx methods in the method javadoc.

File: README.md [hunk 1]
```diff
@@ -6,28 +6,38 @@ See below for [how to install EasyBind in your project](#use-easybind-in-your-pr
...
+## JavaFX 19 - Breaking Changes
+
+JavaFX 19 introduced new methdods `flatMap`, `map`, `orElse`, `filter` which conflict with the methods from EasyBind. In contrast to the EasyBind implementation they do return `null` and not Optional. Therefore the methods from EasyBind have been renamed with the suffix `opt` to avoid conflicts.
+You therefore have to rename the methods in your code to match the renamed.
+
 ## Getting started
 
 The simplest way is to use the `EasyBind.wrap*` methods to create wrappers around standard JavaFX observable values or lists.
-The wrapper then gives you access to all the features of EasyBind.
+The wrapper then gives you access to all the features of EasyBind.a
```
Comment (tobiasdiez) [inline README.md:17]: typo?

File: README.md [hunk 2]
```diff
@@ -178,7 +197,7 @@ EasyBind.includeWhen(edge.getStyleClass(), "highlight", line.hoverProperty());
 
 ### Optional observable values
 
-One often faces the situation that observables take `null` values. 
+One often faces the situation that observables take `null` values.
 The `wrapNullable` provides a wrapper around the observable that provides convenient helper methods similar to the `Optional` class.
 
 ```java
```
Comment (tobiasdiez) [inline README.md:203]: These methods here need to be updated, right?

File: build.gradle [hunk 3]
```diff
@@ -13,7 +13,7 @@ repositories {
 }
 
 group = 'com.tobiasdiez'
-version = '2.2.1-SNAPSHOT'
+version = '3.0.0'
```
Comment (tobiasdiez) [inline build.gradle:16]: Please revert this, I'll do this in a new commit. Thanks!

File: build.gradle [hunk 4]
```diff
@@ -34,7 +34,7 @@ java {
 }
 
 javafx {
-    version = "17"
+    version = "19-ea+10"
```
Comment (tobiasdiez) [inline build.gradle:37]: Please revert this, I don't want this to use an alpha release

File: build.gradle [hunk 5]
```diff
@@ -136,6 +136,11 @@ test {
     }
 }
 
+compileTestJava {
```
Comment (tobiasdiez) [inline build.gradle:139]: Why is this necessary?

File: gradle/wrapper/gradle-wrapper.properties [hunk 6]
```diff
@@ -1,6 +1,5 @@
-#Tue May 05 11:33:00 CEST 2020
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.3-all.zip
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-all.zip
```
Comment (tobiasdiez) [inline gradle/wrapper/gradle-wrapper.properties:3]: Can you please extract the gradle update to a new PR (for cleaner git history). Thanks!

File: src/main/java/com/tobiasdiez/easybind/optional/ObservableOptionalValue.java [hunk 7]
```diff
@@ -91,12 +92,12 @@ default EasyObservableValue<T> asOrdinary() {
 
     /**
      * Returns a new observable that holds the same value as this observable when the value is present and matches the given predicate,
-     * otherwise empty.
+     * otherwise Empty Optional.
```
Comment (tobiasdiez) [inline src/main/java/com/tobiasdiez/easybind/optional/ObservableOptionalValue.java:95]:
```suggestion
     * otherwise an empty optional.
```

**Review by tobiasdiez — CHANGES_REQUESTED** [review]
Thanks for the followup!

File: build.gradle (hunk 5 above)
Comment (Siedlerchr) [reply build.gradle:139]: I think this was a leftover to get it working in eclipse

File: build.gradle (hunk 4 above)
Comment (Siedlerchr) [reply build.gradle]: Well, I needed it to test it against jfx19 which is not yet released to identify the name clashes which is the whole point of this PR. Changed it now to 18

**Comment by Siedlerchr** [comment]
Javafx 19 is here: https://twitter.com/GluonHQ/status/1569648445157146626

File: src/main/java/com/tobiasdiez/easybind/optional/ObservableOptionalValue.java [hunk 8]
```diff
@@ -91,21 +93,25 @@ default EasyObservableValue<T> asOrdinary() {
...
     /**
      * Returns a new observable that holds the same value as this observable when the value is present and matches the given predicate,
-     * otherwise empty.
+     * otherwise it holds an empty optional.
      *
      * @param predicate the predicate to apply to a value, if present
      * @throws NullPointerException if the predicate is {@code null}
      */
-    OptionalBinding<T> filter(Predicate<? super T> predicate);
+    OptionalBinding<T> filterOpt(Predicate<? super T> predicate);
```
Comment (tobiasdiez) [inline src/main/java/com/tobiasdiez/easybind/optional/ObservableOptionalValue.java:101]: I cannot find anything about a new `filter` method in JavaFX 19. Is this really necessary?

**Comment by tobiasdiez** [comment]
For the release notes:
```
## JavaFX 19 - Breaking Changes

JavaFX 19 introduced new methdods `flatMap`, `map`, `orElse`, `filter` which conflict with the methods from EasyBind. In contrast to the EasyBind implementation they do return `null` and not Optional. Therefore the methods from EasyBind have been renamed with the suffix `opt` to avoid conflicts.
You therefore have to rename the methods in your code to match the renamed.
```

File: src/main/java/com/tobiasdiez/easybind/optional/ObservableOptionalValue.java (hunk 8 above)
Comment (Siedlerchr) [reply src/main/java/com/tobiasdiez/easybind/optional/ObservableOptionalValue.java:101]: I aimed for consistency here

Comment (tobiasdiez) [reply src/main/java/com/tobiasdiez/easybind/optional/ObservableOptionalValue.java:101]: Consistency is sadly the thing that we loose here anyway (e.g. there is still an `orElse` method). So please revert this.

Comment (Siedlerchr) [reply src/main/java/com/tobiasdiez/easybind/optional/ObservableOptionalValue.java:101]: The only name clash was about the optional bindings, including orElse

Comment (tobiasdiez) [reply src/main/java/com/tobiasdiez/easybind/optional/ObservableOptionalValue.java:101]: I would prefer if only the methods that really clash with the new javafx methods are renamed.

**Comment by tobiasdiez** [comment]
Thanks!
